contracts/governance/ApprovalReceiver.sol

```diff
@@ -0,0 +1,80 @@
+pragma solidity ^0.5.17;
+
+import "./ErrorDecoder.sol";
+import "../token/IApproveAndCall.sol";
+
+/**
+ * Base contract for receiving approval from SOV token
+ */
+contract ApprovalReceiver is ErrorDecoder, IApproveAndCall {
+	modifier onlyThisContract() {
+		//accepts calls only from receiveApproval function
+		require(msg.sender == address(this), "unauthorized");
+		_;
+	}
+
+	/**
+	 * @notice receives approval from SOV token
+	 * @param _data the data will be used for low level call
+	 */
+	function receiveApproval(
+		address, /*_sender*/
+		uint256, /*_amount*/
+		address, /*_token*/
+		bytes memory _data
```

Comment:
all targets receive `_sender, _amount` as first two arguments. You can check that `_data[4:36] == _sender && _data[36:68] == _amount` (python notation for slices)
You also can check if `_token == msg.sender == _getToken()`

Reply:
done